pipelines/rbsa/tasks/find_sensitivities.py

```diff
@@ -0,0 +1,212 @@
+                enduse_df = zipcode_df[enduse]
+
+                if enduse == 'Heating':
+                    A_heat = A[:,:-1]
+                    x = self.get_baseload(enduse_df, A_heat)
+                    x = np.append(x, [0])
+
+                elif enduse == 'Cooling':
+                    A_cool = np.delete(A, -2, 1)
+                    x = self.get_baseload(enduse_df, A_cool)
+                    cooling_sensitivity = x[-1]
+                    x = np.append(x[:-1], [0])
+                    x = np.append(x, [cooling_sensitivity])
+
+                else:
+                    A_base = A[:,:-2]
+                    x = self.get_baseload(enduse_df, A_base)
+                    x = np.append(x, [0])
+                    x = np.append(x, [0])
+
+                heat_sens = x[-2]
+                cool_sens = x[-1]
+
+                # adjust x by first value
+                x = np.append(x[0],x[1:48]+x[0])
+                x = np.append(x,[heat_sens, cool_sens])
+
+                zipcode_loadshapes[enduse] = x
+
+            zipcode_loadshapes.insert(loc=0, column='zipcode', value=zipcode)
+
+            # print(zipcode_loadshapes.head(4))
+
+
+            loadshapes = loadshapes.append(zipcode_loadshapes)
+
+        self.validate(loadshapes)
+        self.on_complete({self.output_artifact_loadshapes: loadshapes})
+
+    def get_baseload(self, df, A):
+        """ This function gets baseload for non weather sensitive enduses
+        """
+
+        At = A.transpose()
+        y = np.asarray(df).transpose()
+        M = np.matmul(np.linalg.inv(np.matmul(At,A)),At)
+        x = np.matmul(M,y)
+
+        return x
+
+    def get_nonsensitive_A(self, weather):
+        """constructs A matrix for non weather sensitive loads
+        """
+
+        A = np.zeros((len(weather.index),48),float)
+
+        ts = datetime.datetime(weather.index[0].year,1,1,0,0,0)
+        dt = datetime.timedelta(hours=1) 
+
+        for h in range(len(weather.index)):
+            A[h][0] = 1
+            hh = h%24
+            
+            if weather.index[h].weekday() < 5:
+                A[h][hh] = 1.0
+            else:
+                A[h][hh+24] = 1.0
+
+            ts += dt
+
+        return A
+
+    def get_A(self, weather):
+        """constructs A matrix for non weather sensitive loads
+        """
+
+        A = np.zeros((len(weather.index),50),float)
+
+        ts = datetime.datetime(weather.index[0].year,1,1,0,0,0)
+        dt = datetime.timedelta(hours=1) 
+
+        for h in range(len(weather.index)):
+            A[h][0] = 1
+            hh = h%24
+            
+            if weather.index[h].weekday() < 5:
+                A[h][hh] = 1.0
+            else:
+                A[h][hh+24] = 1.0
+
+            if weather['Temperature'][h] < self.theat:
+                A[h][(24*2)] = weather['Temperature'][h]-self.theat
+            elif weather['Temperature'][h] > self.tcool:
+                A[h][(24*2)+1] = weather['Temperature'][h]-self.tcool
```

Comment:
same whitespace issue around all the operators above

Reply:
fixed